### ISSUE TITLE
keep RTF formatting when appending RTF files

### ISSUE THREAD

**Issue by dylanjohnlukes** [Merck/r2rtf#128]
Hello, 

I currently have 10 separate but identically formatted RTF files I'd like to append into one master RTF file. I am currently able to append without errors, but after the first page the table body text reverts back to default font settings in the master RTF file. In addition, the header and footer text automatically revert back to default font settings starting page 1. 

Any help is much appreciated. Importantly, each separate RTF file is already created and properly formatted. I've checked each one individually before running the append loop and are exactly as I want them pre-appending. 

**RTF Assemble Function (https://r4csr.org/)** 
rtf_assemble1 <- function(path) {
  rtf <- lapply(path, readLines)
  
  n <- length(rtf)
  start <- c(1, rep(2, n - 1))
  end <- vapply(rtf, length, numeric(1))
  end[-n] <- end[-n] - 1
  
  for (i in 1:n) {
    rtf[[i]] <- rtf[[i]][start[i]:end[i]]
    if (i < n) rtf[[i]] <- c(rtf[[i]], r2rtf:::as_rtf_new_page())
  }
  rtf <- do.call(c, rtf)
  
  rtf
}

**Comment by elong0527** [comment]
could you attach a few tables that can reproduce the issue?

Please also try the new function in r2rtf, we will need to update r4csr on this topic. Thanks!

https://merck.github.io/r2rtf/reference/assemble_rtf.html

**Comment by dylanjohnlukes** [comment]
sure here is a dummy example: 
(1) example_1.rtf has correct formatting (i.e., Arial font both body, header, footer text) 
(2) example_2.rtf has correct formatting (i.e., Arial font both body, header, footer text) 
(3) example_combined.rtf (use assemble_rtf() to combine example1.rtf and example2.rtf) no longer has correct formatting 

Interestingly, the body text is correct (i.e., Arial font) on the first page but reverts to Courier New on the second page. All header and footer font text immediately revert to Courier New. 

"example_1" 
![example_1](https://user-images.githubusercontent.com/113366162/207644857-e05bf213-745d-4223-8081-37e94547ed3f.JPG)

"example_2" 
![example_2](https://user-images.githubusercontent.com/113366162/207644871-a8b1cd0c-bae3-4baa-9b72-c27edbec3399.JPG)

"example_combined" 
![example_combined](https://user-images.githubusercontent.com/113366162/207644875-bc362d86-e553-497b-b011-637a451a9858.JPG)

**Comment by fb-elong** [comment]
The code below works as expected. 

Table body are both in Courier  font from my end. 

Could you try the code and see if it works? 

For me to find the root cause, please share code that can fully reproduce the issue from your end.

```
library(r2rtf)
library(magrittr)

file <- replicate(2, tempfile(fileext = ".rtf"))
file1 <- head(iris) %>% rtf_body(text_font = 6) %>% rtf_encode() %>% write_rtf(file[1])
file2 <- head(cars) %>% rtf_page(orientation = "landscape") %>%
              rtf_body(text_font = 6) %>% rtf_encode() %>% write_rtf(file[2])
output <- tempfile(fileext = ".rtf")

assemble_rtf(
  input = file,
  output = "tmp.rtf"
)
```

**Comment by dylanjohnlukes** [comment]
No problem. See code below. You need to change the "text_font" to something other than Courier (e.g., text_font = 3 below). Courier appears to be the default font the program is reverting to when RTF tables are combined. 

# load packages
library(r2rtf)      # Reporting in RTF format
library(magrittr)   # Required for %>% 

# example  
file <- replicate(2, tempfile(fileext = ".rtf"))

file1 <- head(iris) %>% 
  # table header 
  rtf_page_header(
    text =  c("Testing Testing Testing"), 
    text_font = 3,
    text_font_size = 10
  ) %>%
  # table footer 
  rtf_page_footer(
    text = c("Testing Testing Testing"),
    text_font = 3,
    text_font_size = 10
  ) %>% 
  # table body 
  rtf_body(
    text_font = 3, 
    text_font_size = 10
  ) %>% 
  
  # encode and write
  rtf_encode() %>% 
  write_rtf(file[1])

file2 <- head(cars) %>% 
  # page orientation 
  rtf_page(orientation = "landscape") %>%
  # table header 
  rtf_page_header(
    text =  c("Testing Testing Testing"), 
    text_font = 3,
    text_font_size = 10
  ) %>%
  # table footer 
  rtf_page_footer(
    text = c("Testing Testing Testing"),
    text_font = 3,
    text_font_size = 10
  ) %>% 
  # table body 
  rtf_body(
    text_font = 3, 
    text_font_size = 10
  ) %>% 
  # encode and write 
  rtf_encode() %>% 
  write_rtf(file[2])

output <- tempfile(fileext = ".rtf") 

assemble_rtf(
  input = file,
  output = "tmp.rtf"
)

**Comment by dylanjohnlukes** [comment]
No problem. See code below. You need to change the "text_font" to something other than Courier (e.g., text_font = 3 below). Courier appears to be the default font the program is reverting to when RTF tables are combined.

library(r2rtf)      # Reporting in RTF format
library(magrittr)   # Required for %>% 

file <- replicate(2, tempfile(fileext = ".rtf"))

file1 <- head(iris) %>% 
  rtf_page_header(
    text =  c("Testing Testing Testing"), 
    text_font = 3,
    text_font_size = 10
  ) %>%
  rtf_page_footer(
    text = c("Testing Testing Testing"),
    text_font = 3,
    text_font_size = 10
  ) %>% 
  rtf_body(
    text_font = 3, 
    text_font_size = 10
  ) %>% 
  rtf_encode() %>% 
  write_rtf(file[1])

file2 <- head(cars) %>% 
  rtf_page(orientation = "landscape") %>%
  rtf_page_header(
    text =  c("Testing Testing Testing"), 
    text_font = 3,
    text_font_size = 10
  ) %>%
  rtf_page_footer(
    text = c("Testing Testing Testing"),
    text_font = 3,
    text_font_size = 10
  ) %>% 
  rtf_body(
    text_font = 3, 
    text_font_size = 10
  ) %>% 
  rtf_encode() %>% 
  write_rtf(file[2])

output <- tempfile(fileext = ".rtf") 

assemble_rtf(
  input = file,
  output = "tmp.rtf"
)

**Comment by fb-elong** [comment]
The issue is about this line 

https://github.com/Merck/r2rtf/blob/master/R/assemble.R#L64

that needs to be updated as below to avoid duplicate font claim in RTF. Will create a PR to fix the issue.  

```
start <- c(1, vapply(rtf[-1], function(x) max(grep("fcharset",rtf[[1]])) + 2, numeric(1)))
```

**Comment by nanxstats** [comment]
Fixed by #129
